routing/pathfind.go

```diff
@@ -630,22 +634,38 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 
 	// Build a preliminary destination hop structure to obtain the payload
 	// size.
-	var mpp *record.MPP
-	if r.PaymentAddr != nil {
-		mpp = record.NewMPP(amt, *r.PaymentAddr)
-	}
+	var finalHop route.Hop
+	if r.BlindedPayment != nil {
```

Comment:
+1

Reply:
Another benefit of adding an [interface](https://github.com/lightningnetwork/lnd/pull/7852/#pullrequestreview-1729181183) is that we could add a `finalHopSize(blindedPayment, restrictParams)` method as well.

Reply:
great idea 👌

Reply:
So pulled this logic out in its own function : `lastHopPayloadSize` but did not go for the integration into the `AdditionalEdge` interface. One reason was that we also need the lasthop calculation for non-blinded hops which would leave our interface with a bad name I guess ? Apart from that I did not see a lot of advantages because the function signatures would need to be the same for an interface although both `Private` and `Blinded` edges do need different arguments. So we see it already in the `PayloadSize` function that this change is not optimal here because for the blinded path we just carry around the arguments of the function without ever needing them:
```

func (edge *BlindedEdge) PayloadSize(_ lnwire.MilliSatoshi, _ uint32,
	_ bool, _ uint64) uint64 {
	hop := route.Hop{
		BlindingPoint: edge.blindingPoint,
		LegacyPayload: false,
		EncryptedData: edge.cipherText,
	}
	// For blinded paths the next chanID is in the encrypted data tlv.
	return hop.PayloadSize(0)
}
```